### PR TITLE
:tada: Adds Drift Detection Demo

### LINKED ISSUE #350: BUG-Data Node Viewer does not update on scenario execution
**Description**
The Data Node Viewer does not update itself when a scenario is executed again. This is problematic when the user runs a scenario, changes the inputs, and re-runs it. The Data Node Viewer will show data from the previous execution, and the user has to refresh the page to see the new result.

**How to reproduce**
Use the following code:
[core.zip](https://github.com/Avaiga/taipy-gui/files/13113593/core.zip)

1. On Taipy 3.0.0.dev3
2. Run app.py
3. Create a scenario using the scenario_selector
image
image
4. Select the scenario and run it
image
5. Check the value in the Data Node Viewer for "evaluation" (should be a random float)
image
6. Run the scenario again. The Data Node Viewer will still show the same value until you refresh the page

**Expected behavior**
Executing a scenario should update the Data Node Viewer elements with the data of the most recent execution

**Runtime environment**
 - OS: Windows 11
 - Browser: Chrome
 - Taipy 3.0.0.dev3

### PR DESCRIPTION
**Pull request by AlexandreSajus**
Adds a demo about showcasing the ability to select inputs and execute and visualize outputs of data pipelines in a Taipy application using the example of detecting drift on a dataset.

image

(the live demo will have issues until the following issue is fixed: https://github.com/Avaiga/taipy/issues/350)

### CODE REVIEW

File: docs/knowledge_base/demos/drift_detection.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/drift_detection.md]: Please try breaking lines after 100 characters to make it easier to review. Thx.
```suggestion
This demo aims to showcase the ability to select inputs and execute and visualize outputs of
data pipelines in a Taipy application. Here, we use the example of detecting drift on a
dataset.
```

File: docs/knowledge_base/demos/drift_detection.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/drift_detection.md]:
```suggestion
Data drift is a concept used mainly in machine learning, where inference data distribution strays from training data distribution. Various factors, such as changes in the underlying data
source, changes in the data collection process, or changes in the data storage can cause data 
to drift. This generally causes a performance issue called training-serving skew, where the
model used for inference is not used for the distribution of the inference data and fails to
generalize.
```

File: docs/knowledge_base/demos/drift_detection.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/drift_detection.md]:
```suggestion
Statistical tests exist to detect drift in a dataset. These tests calculate the probability of
two series coming from the same distribution. If the probability is under a threshold, we
consider that there is drift.
```

File: docs/knowledge_base/demos/drift_detection.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/drift_detection.md]:
```suggestion
The dataset used in this app has both numerical (age, body mass index, blood pressure) and
categorical (sex) columns. That is why we start by separating these columns first. Then, we run
a statistical test for each column to detect drift (Kolmogorov for numerical, Chi-squared for
categorical).
```

File: docs/knowledge_base/demos/drift_detection.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/drift_detection.md]:
```suggestion
# How to use the Application
```

File: docs/knowledge_base/demos/drift_detection.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/drift_detection.md:1]: The screenshots are too small to be easily readable. Can you reduce the size of the windows to reduce their width? It should help a lot.

File: docs/knowledge_base/demos/drift_detection.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/drift_detection.md]:
```suggestion
Here, we are selecting data_big, a dataset similar to the dataset of reference but with rows with higher blood pressure values. We see on the blood pressure distribution chart that the
distribution of the comparison dataset in red is shifted to the right compared to the
reference dataset in green.
```

File: docs/knowledge_base/demos/drift_detection.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/drift_detection.md]:
```suggestion
Here, we see that the p-value for the Kolmogorov test of the blood pressure column is under
0.05, which means that the probability of both datasets of blood pressure coming from the same
distribution is under 5%. We can reject the hypothesis that both datasets come from the same
distribution and conclude that there is a drift in the blood pressure column.
```

File: docs/knowledge_base/demos/drift_detection.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/drift_detection.md]:
```suggestion
```

File: docs/knowledge_base/demos/index.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/index.md]:
```suggestion
        <p> Showcases the ability to select inputs and execute and visualize outputs of data
            pipelines in a Taipy application by using the example of detecting drift on a
            dataset.
```

File: docs/knowledge_base/index.md
Comment (jrobinAV) [inline docs/knowledge_base/index.md]:
```suggestion
        <p> Showcases the ability to select inputs and execute and visualize outputs of data
            pipelines in a Taipy application by using the example of detecting drift on a
            dataset.
```

File: docs/knowledge_base/index.md
Comment (jrobinAV) [inline docs/knowledge_base/index.md]:
```suggestion
  <li data-keywords="gui ai scenario datanode dag configuration">
```

File: docs/knowledge_base/demos/index.md
Comment (jrobinAV) [inline docs/knowledge_base/demos/index.md]:
```suggestion
  <li class="tp-col-12 tp-col-md-6 d-flex" data-keywords="gui ai scenario datanode dag configuration">
```

**Review by jrobinAV — COMMENTED** [review]
1. This is minor, but I believe we can improve the UX by displaying the scenario viewer and the dag right below the input data nodes selections.

2. The scenario viewer does not need to be expanded nor expandable.

3. You did not created a more realistic use case with monthly scenarios with cumulative drift month after month as discussed. I am more than fine with that to make the demo easily understandable. However, Please consider having an advanced version of this demo to first exhibit the cycle usage and second to show how Taipy helps end users managing data overtime. It is part of our will to position Taipy as a tool to help data management operations in production and industrial context.

**Comment by AlexandreSajus** [comment]
> 1. This is minor, but I believe we can improve the UX by displaying the scenario viewer and the dag right below the input data nodes selections.
> 2. The scenario viewer does not need to be expanded nor expandable.
> 3. You did not created a more realistic use case with monthly scenarios with cumulative drift month after month as discussed. I am more than fine with that to make the demo easily understandable. However, Please consider having an advanced version of this demo to first exhibit the cycle usage and second to show how Taipy helps end users managing data overtime. It is part of our will to position Taipy as a tool to help data management operations in production and industrial context.

1. I feel that displaying data and data distribution after selecting the data source makes more sens as you can see the changes without scrolling
2. Fixed now
3. Sure, I did not see how to fit this in this demo without making it too complex. I will work on another demo about cycle and data management

**Comment by jrobinAV** [comment]
> 1. I feel that displaying data and data distribution after selecting the data source makes more sens as you can see the changes without scrolling
> 2. Fixed now
> 3. Sure, I did not see how to fit this in this demo without making it too complex. I will work on another demo about cycle and data management

Sounds good!

**Comment by jrobinAV** [comment]
Any update on this one?

**Comment by AlexandreSajus** [comment]
> Any update on this one?

Yeah I fixed what you asked, just waiting for approval

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**
